### ISSUE TITLE
Last updated date appears to be incorrect

### ISSUE THREAD

**Issue by michaelmior** [actions/stale#1184]
**Description:**
As the excerpt from the log below shows, I have a pull request that is identified as having been updated within the past several days even though the last update I can see is nearly one year ago.

```
[#514] Pull request #514
  [#514] Found this pull request last updated at: 2024-10-15T01:53:46Z
```

**Action version:**
v9

**Platform:**
- [x] Ubuntu
- [ ] macOS
- [ ] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Repro steps:**  
I am unsure how to reproduce this issue with a fresh repository, but [here is an example](https://github.com/apache/calcite/actions/runs/11377902079/job/31652795554) of a run of this action that exhibits the problem.
For example, look at [pull request 512](https://github.com/apache/calcite/pull/514).
It has not been updated in more than a year, but the action seems to think it was updated within the past several days.

**Expected behavior:**
The correct date of the last update should be identified.

**Actual behavior:**
The date of last update is not identified correctly.

**Comment by michaelmior** [comment]
I do see the `updated_at` date returned by the API is within the past several days, so I assume this is the root of the issue.

**Comment by HarithaVattikuti** [comment]
Hello @michaelmior 👋,
Thank you for reporting this issue. We will investigate it and get back to you as soon as we have some feedback.

**Comment by dktapps** [comment]
Yep, running into this with a bunch of stale PRs on my repo too. For some reason a bunch of them got unmarked as stale 3 days ago.

**Comment by Adam262Alloy** [comment]
I am noticing this in our repos as well. Taking three example PRs for example, they each have a comment from the Stale Github action that they were quickly marked and unmarked as stale on 1/6/25. Neither PR has user activity on that day (they are automated dependabot type PRs), and all should be stale in that they are older than our configured threshhold for staleness.

![image](https://github.com/user-attachments/assets/9285ab50-2bed-4d90-ac09-d058c4210bf8)

When I look in the Github action logs, I can see the action processed these PRs consecutively in the same batch, and they have very similar updated at messages:

PR 1 --  `Found this pull request last updated at: 2025-01-06T22:22:13Z`
PR 2 --  `Found this pull request last updated at: 2025-01-06T22:22:13Z`
PR 3 --  `Found this pull request last updated at: 2025-01-06T22:22:14Z`

**Comment by wandbjake** [comment]
Yeah, same thing when I tried on my repo. I think that the act of adding the Stale tag updates the updated_at. Github must've changed this internally, but this definitely breaks the logic. The first pass through it adds the tag, then the second pass it removes it.

**Comment by mala-sap** [comment]
Yes, can confirm, the same finding here: the action adds the label Stale at the first run, then the second run removes the label again, despite there was no change at all at the PR:

![Image](https://github.com/user-attachments/assets/47b31981-8bcc-42b7-b8b3-324b6422b2b4)

![Image](https://github.com/user-attachments/assets/9d62b67a-1afa-4e46-a2f6-ddcf99bd1b30)

**Comment by chiranjib-swain** [comment]
Hi @mala-sap, thanks for sharing these findings .

<img width="927" height="292" alt="Image" src="https://github.com/user-attachments/assets/5ccaa0f7-b10e-4ebc-90e8-82b54cbdb930" />

**Image 1:**
 The `stale` label was added and then removed again, even though there was no visible PR change. To help us debug, could you please share the relevant action logs + a minimal reproducible example (or temporary repo access)? That will confirm whether this is config-related or a bug. This looks quite similar to the main user’s issue here, so comparing both cases may help narrow down the root cause.

<img width="720" height="560" alt="Image" src="https://github.com/user-attachments/assets/40f46984-d0b3-42b7-bff8-f4908de5f0a6" />

**Image 2:**
 This looks like expected behavior: the PR went inactive → `stale` applied → SonarQube bot ran ✅ → since `remove-stale-when-updated: true` is enabled, that bot update triggered `stale` removal.

**Comment by valentijnscholten** [comment]
The problem seems to be that when the bot adds a label, the `update_at` timestamp changes. So on the next run the label is removed again.

**Comment by shamoon** [comment]
First, thanks for this tool, its incredibly valuable to us.

I think the above comment is correct, the bot is counting its own changes as altering the last updated, and it seems like a regression (see #1008). And the same in our case, there no appreciable updates except the bot adding and removing 'stale' (twice!), most recently yesterday:

<img width="590" height="124" alt="Image" src="https://github.com/user-attachments/assets/d96e2c61-af97-4923-9a2f-746653ed224b" />

Action run from yesterday:

```
[#11353] Pull request #11353
  [#11353] Found this pull request last updated at: 2025-12-08T23:01:57Z
  [#11353] The option only-labels (​[https://github.com/actions/stale#only-labels​)](https://github.com/actions/stale#only-labels%E2%80%8B)) was not specified
  [#11353] └── Continuing the process for this pull request
  [#11353] Days before pull request stale: 14
  [#11353] The pull request is not closed nor locked. Trying to remove the close label...
  [#11353] ├── The close-pr-label (​[https://github.com/actions/stale#close-pr-label​)](https://github.com/actions/stale#close-pr-label%E2%80%8B)) option was not set
  [#11353] └── Skipping the removal of the close label
  [#11353] This pull request does not include a stale label
  [#11353] The option any-of-labels (​[https://github.com/actions/stale#any-of-labels​)](https://github.com/actions/stale#any-of-labels%E2%80%8B)) was not specified
  [#11353] └── Continuing the process for this pull request
  [#11353] This pull request has no milestone
  [#11353] └── Skip the milestones checks
  [#11353] This pull request has no assignee
  [#11353] └── Skip the assignees checks
  [#11353] This pull request is not stale
  [#11353] The option ignore-updates (​[https://github.com/actions/stale#ignore-updates​)](https://github.com/actions/stale#ignore-updates%E2%80%8B)) is disabled. The stale counter will take into account updates and comments on this pull request to avoid to stale when there is some update
  [#11353] This pull request should be stale based on the last update date the 08-12-2025 (2025-12-08T23:01:57Z)
  [#11353] This pull request should be marked as stale based on the option days-before-pr-stale (​[https://github.com/actions/stale#days-before-pr-stale​)](https://github.com/actions/stale#days-before-pr-stale%E2%80%8B)) (14)
  [#11353] Marking this pull request as stale
  [#11353] This pull request is now stale
  [#11353] This pull request is already stale
  [#11353] Checking for label on this pull request
  [#11353] Pull request marked stale on: 2025-12-23T03:32:17Z
  [#11353] Checking for comments on pull request since: 2025-12-23T03:32:17Z
  [#11353] Comments that are not the stale comment or another bot: 0
  [#11353] Pull request has been commented on: false
  [#11353] Days before pull request close: 7
  [#11353] The option remove-stale-when-updated (​[https://github.com/actions/stale#remove-stale-when-updated​)](https://github.com/actions/stale#remove-stale-when-updated%E2%80%8B)) is: true
  [#11353] The stale label should not be removed
  [#11353] marked stale this run, so don't check for updates
  [#11353] Pull request has been updated since it was marked stale: false
  [#11353] Pull request has been updated in the last 7 days: true
  [#11353] Stale pull request is not old enough to close yet (hasComments? false, hasUpdate? true)
  [#11353] 3 operations consumed for this pull request
```

Action run from today:

```
[#11353] Pull request #11353
  [#11353] Found this pull request last updated at: 2025-12-23T03:32:45Z
  [#11353] The option only-labels (​[https://github.com/actions/stale#only-labels​)](https://github.com/actions/stale#only-labels%E2%80%8B)) was not specified
  [#11353] └── Continuing the process for this pull request
  [#11353] Days before pull request stale: 14
  [#11353] The pull request is not closed nor locked. Trying to remove the close label...
  [#11353] ├── The close-pr-label (​[https://github.com/actions/stale#close-pr-label​)](https://github.com/actions/stale#close-pr-label%E2%80%8B)) option was not set
  [#11353] └── Skipping the removal of the close label
  [#11353] This pull request includes a stale label
  [#11353] The option any-of-labels (​[https://github.com/actions/stale#any-of-labels​)](https://github.com/actions/stale#any-of-labels%E2%80%8B)) was not specified
  [#11353] └── Continuing the process for this pull request
  [#11353] This pull request has no milestone
  [#11353] └── Skip the milestones checks
  [#11353] This pull request has no assignee
  [#11353] └── Skip the assignees checks
  [#11353] This pull request is already stale
  [#11353] Checking for label on this pull request
  [#11353] Pull request marked stale on: 2025-12-23T03:32:17Z
  [#11353] Checking for comments on pull request since: 2025-12-23T03:32:17Z
  [#11353] Comments that are not the stale comment or another bot: 0
  [#11353] Pull request has been commented on: false
  [#11353] Days before pull request close: 7
  [#11353] The option remove-stale-when-updated (​[https://github.com/actions/stale#remove-stale-when-updated​)](https://github.com/actions/stale#remove-stale-when-updated%E2%80%8B)) is: true
  [#11353] The stale label should not be removed
  [#11353] Pull request has been updated since it was marked stale: true
  [#11353] Remove the stale label since the pull request has been updated and the workflow should remove the stale label when updated
  [#11353] The pull request is no longer stale. Removing the stale label...
  [#11353] Removing the label "stale" from this pull request...
  [#11353] The label "stale" was removed
  [#11353] Skipping the process since the pull request is now un-stale
  [#11353] 3 operations consumed for this pull request
```

**Comment by shamoon** [comment]
I've figured out the cause, at least in our case:

> ```
>   [#11353] Found this pull request last updated at: 2025-12-23T03:32:45Z
>   ...
>   [#11353] Pull request marked stale on: 2025-12-23T03:32:17Z
>   [#11353] Checking for comments on pull request since: 2025-12-23T03:32:17Z
>   ...
>   [#11353] Pull request has been updated since it was marked stale: true
>   [#11353] Remove the stale label since the pull request has been updated and the workflow should remove the stale label when updated
> ```

and then:

https://github.com/actions/stale/blob/a21a0816299b11691f9592ef0d63d08e02f06d9d/src/classes/issues-processor.ts#L747-L751

is checking to see if the dates are the same with a 15 second tolerance, but in this case, it was 28 seconds between the events.

Obviously I dont know the *ideal* tolerance, but I propose increasing it a bit to 30 seconds in #1310

**Comment by chiranjib-swain** [comment]
Hi all — @michaelmior @Adam262Alloy @wandbjake @mala-sap @valentijnscholten
There’s a proposed fix in [@actions/stale/pull/1311](https://github.com/actions/stale/pull/1311) (“Enhancement: ignore stale labeling events”) that may address the incorrect “last updated” date behavior discussed here.
Could you please try testing this on your side and let us know if it resolves the issue for your repo/workflow? To test, you can point your workflow to the PR ref, for example:
`- uses: actions/stale@refs/pull/1311/head`



Thanks very much for taking a look!

**Comment by chiranjib-swain** [comment]
Thanks, @mala-sap, for testing this.

The error you encountered is not unexpected in certain environments. Using uses: `actions/stale@refs/pull/1311/head` attempts to fetch a Pull Request (PR) reference, which can be blocked by repository/organization policies.

To simplify testing, I've pushed the changes from PR #1311 to a branch in this repository: `enhancement/ignore-stale-labeling-events`.

Could you try:
`- uses: actions/stale@enhancement/ignore-stale-labeling-events`
and let us know if it fixes the “last updated” behavior?
 Your feedback is greatly appreciated.

**Comment by ghost** [comment]
hi all - @michaelmior , @valentijnscholten , @Adam262Alloy , @wandbjake 
i would like to encourage you to test the fix provided by @chiranjib-swain . happy to get your feedback on this.

**Comment by chiranjib-swain** [comment]
Hi all - @michaelmior, @valentijnscholten, @Adam262Alloy, @wandbjake

Could you please try out the fix above and let us know whether it resolves the issue? Thanks in advance for any feedback.

**Comment by mikebryant** [comment]
Heya, we've been experiencing this issue where a lot of PRs were marked stale, and then had it removed. This happened repeatedly for some PRs.

We switched to `uses: actions/stale@enhancement/ignore-stale-labeling-events` a couple of weeks ago, and we've not seen the problem reoccur since :)

**Comment by chiranjib-swain** [comment]
Thanks for the report and the detailed info.

This is fixed on `main` via actions/stale#1311 and will be included in the next release. Please re-test once the next version is out and reopen/comment if you still see the issue.

Closing as resolved by actions/stale#1311.

**Comment by shamoon** [comment]
@chiranjib-swain any ETA on when the next release will be out, still getting snagged by this bug on 10.2.0:

<img width="641" height="121" alt="Image" src="https://github.com/user-attachments/assets/42d6a348-0342-4134-b2a1-07d38c18bbd3" />

Thank you
